tutorial/04_embedding_net.ipynb

```diff
@@ -0,0 +1,385 @@
+    "        \n",
+    "    def forward(self, x):\n",
+    "        x = x.view(-1, 1, 32, 32)\n",
+    "        x = self.pool(F.relu(self.conv1(x)))\n",
+    "        x = x.view(-1, 6*4*4)\n",
+    "        x = F.relu(self.fc(x))\n",
+    "        return x\n",
+    "\n",
+    "embedding_net = SummaryNet()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## The inference procedure\n",
+    "\n",
+    "With the summary features neural net defined and instantiated, we can follow the usual workflow of an inference procedure in `sbi`. The `embedding_net` object appears as an input argument when instantiating the neural density estimator on `utils.posterior_nn`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "0fb2487131e9401cb63d5561e912287c",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "HBox(children=(FloatProgress(value=0.0, description='Running 10000 simulations.', max=10000.0, style=ProgressS…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "Neural network successfully converged after 132 epochs.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# set prior distribution for the parameters \n",
+    "prior = utils.BoxUniform(low=torch.tensor([0.0, 0.0]), \n",
+    "                             high=torch.tensor([1.0, 2*np.pi]))                           \n",
+    "\n",
+    "# make a SBI-wrapper on the simulator object for compatibility\n",
+    "simulator_wrapper, prior = inference.prepare_for_sbi(simulator_model, prior)\n",
+    "\n",
+    "# instantiate the neural density estimator\n",
+    "neural_posterior = utils.posterior_nn(model='maf', \n",
+    "                                      embedding_net=embedding_net,\n",
+    "                                      hidden_features=10,\n",
+    "                                      num_transforms=2)\n",
+    "\n",
+    "# setup the inference procedure with the SNPE-C procedure\n",
+    "inference = inference.SNPE(simulator_wrapper, prior, \n",
+    "                           density_estimator=neural_posterior, \n",
+    "                           show_progress_bars=True)\n",
+    "\n",
+    "# run the inference procedure on one round and 10000 simulated data points\n",
+    "posterior = inference(num_simulations=10000)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualizing the results\n",
+    "\n",
+    "We now generate 50000 samples of the posterior distribution of $r$ and $\\theta$ when observing an input data point $x$ generated from the `simulator model` with $r = 0.75$ and $\\theta = \\pi/4$. The figure below shows the statistics of these samples."
```

Comment:
I would suggest to split this cell into two cells. Generate `x_observed` in the first cell and then sample and visualize the posterior in the second cell.